### PR TITLE
go/store/util/verbose: Add ability to override Log function.

### PR DESCRIPTION
**Pull request by reltuk**

### CODE REVIEW

**Comment by reltuk** [comment]
@bheni Ok, just adopted `zap.Logger` natively. PTAL.

File: go/store/util/verbose/verbose.go [hunk 1]
```diff
@@ -36,35 +39,30 @@ var (
...
+			return zapcore.DebugLevel.Enabled(l)
 		} else {
-			log.Println(format)
+			return zapcore.WarnLevel.Enabled(l)
 		}
+	})
+	encoder := zapcore.NewConsoleEncoder(zap.NewDevelopmentEncoderConfig())
+	core := zapcore.NewCore(encoder, zapcore.Lock(os.Stderr), enabler)
+	Logger = func(ctx context.Context) *zap.Logger {
+		return zap.New(core)
```
Comment (bheni) [inline go/store/util/verbose/verbose.go]: New logger for each call for the default dolt client?

Comment (bheni) [reply go/store/util/verbose/verbose.go]: Should we set the default log level to error for dolt?

Comment (reltuk) [reply go/store/util/verbose/verbose.go]: Warn seems fine...there is currently no logging though.

**Pull request merged**